src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
+const HEALTH_DEFAULT = 100;
+
 class Animal {
-  // write your code here
+  constructor(name = '') {
+    this.health = HEALTH_DEFAULT;
+    this.name = name;
+    Animal.alive.push(this);
+  }
 }
+Animal.alive = [];
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name = '') {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name = '') {
+    super(name);
+  }
+
+  bite(victim) {
+    if (victim instanceof Herbivore && !victim.hidden) {
+      victim.health -= 50;
+
+      if (victim.health <= 0) {
+        Animal.alive = Animal.alive.filter(
```

Comment:
will be prettier:
```
Animal.alive = Animal.alive.filter(() => {
  return animal => animal.name !== victim.name;
});
```

Reply:
also, it's better to create a static method for the class that will delete the animal:
```
if (victim.health <= 0) {
  Animal.kill(victim);
}
```